### PR TITLE
All doc updates forv15.5

### PR DESCRIPTION
**Pull request by flarnie**
To make this update easy all the doc updates are condensed into this one PR. The smaller other PRs were for ease of review.

This PR includes:
- Changes from ['Update docs for test utils add ons change'](https://github.com/facebook/react/pull/9331) which was approved
- Changes from ['Update docs to deprecate React<CSS>TransitionGroup'](https://github.com/facebook/react/pull/9336)
- Changes from ['Remove `React.createClass` and `React.propTypes` from the docs'](https://github.com/facebook/react/pull/9355)
- Two new commits removing `React.addons` from the docs

To review please just look at the more recent diffs:
- [ ] Review the ['Remove `React.createClass` and `React.propTypes` from the docs'](https://github.com/facebook/react/pull/9355)
- [ ] Review the top two commits; "Remove 'addons' items from left nav, and deprecate 'addons' doc page" and 'Remove references to `React.addons` from docs'

### CODE REVIEW

**Comment by bvaughn** [comment]
Thanks @flarnie !

**Comment by gaearon** [comment]
We need to hold off syncing docs until 15.5 is out right?

**Comment by acdlite** [comment]
Yes

File: docs/docs/addons.md [hunk 1]
```diff
@@ -24,11 +26,12 @@ The add-ons below are considered legacy and their use is discouraged.
...
 ### Deprecated Add-ons
 
-[`LinkedStateMixin`](two-way-binding-helpers.html) has been deprecated.
+- [`LinkedStateMixin`](two-way-binding-helpers.html) has been deprecated.
+- [`TransitionGroup` and `CSSTransitionGroup`](animation.html) have been deprecated.
 
 ## Using React with Add-ons
 
-If using npm, you can install the add-ons individually from npm (e.g. `npm install react-addons-test-utils`) and import them:
+You can install the add-ons individually from npm (e.g. `npm install react-addons-create-fragment`) and import them:
```
Comment (gaearon) [inline docs/docs/addons.md]: Would it make sense to change snippet below to match package name?

File: docs/docs/components-and-props.md [hunk 2]
```diff
@@ -12,8 +12,6 @@ redirect_from:
   - "docs/transferring-props-zh-CN.html"
   - "tips/props-in-getInitialState-as-anti-pattern.html"
   - "tips/communicate-between-components.html"
-prev: rendering-elements.html
```
Comment (gaearon) [inline docs/docs/components-and-props.md]: Why did we remove these? Their ordering seemed important to me.

File: docs/docs/composition-vs-inheritance.md [hunk 3]
```diff
@@ -3,8 +3,6 @@ id: composition-vs-inheritance
 title: Composition vs Inheritance
 permalink: docs/composition-vs-inheritance.html
 redirect_from: "docs/multiple-components.html"
-prev: lifting-state-up.html
-next: thinking-in-react.html
```
Comment (gaearon) [inline docs/docs/composition-vs-inheritance.md]: Same question here.

File: docs/docs/conditional-rendering.md [hunk 4]
```diff
@@ -2,8 +2,6 @@
 id: conditional-rendering
 title: Conditional Rendering
 permalink: docs/conditional-rendering.html
-prev: handling-events.html
```
Comment (gaearon) [inline docs/docs/conditional-rendering.md]: And here.

Comment (flarnie) [inline docs/docs/conditional-rendering.md]: My thought was that they are no longer part of an ordered navigation - each one is now either a stand-alone package, or deprecated. They soon won't live in the main React documentation. The 'next' and 'prev' links only make sense when they are all part of the set of 'addons', but the concept of the set itself is deprecated.

File: docs/docs/addons.md (hunk 1 above)
Comment (flarnie) [inline docs/docs/addons.md]: Yes! Will submit a follow-up PR.
nice catch

File: docs/docs/conditional-rendering.md (hunk 4 above)
Comment (gaearon) [inline docs/docs/conditional-rendering.md]: Hmm. I'm a bit confused. These pages (e.g. Conditional Rendering) don't have to do with addons as far as I can see.

File: docs/docs/context.md [hunk 5]
```diff
@@ -4,11 +4,15 @@ title: Context
 permalink: docs/context.html
 ---
 
+>Note:
+> As of React v15.5 the `React.PropTypes` helper is deprecated, and we recommend using the [`prop-types` library](https://github.com/aackerman/PropTypes) to define `contextTypes`.
```
Comment (bvaughn) [inline docs/docs/context.md]: Let's change the `prop-types` link to https://www.npmjs.com/package/prop-types?

File: docs/docs/conditional-rendering.md (hunk 4 above)
Comment (flarnie) [inline docs/docs/conditional-rendering.md]: You're right - I was just going through the list of add-ons docs and accidentally updated some others too. Thanks! Fixing in that follow-up PR.

**Comment by gaearon** [comment]
Here is what the PropTypes deprecation looks like in the docs today:

The (unintentional) bold is super scary, and there is no mention of the codemod. Somebody reading this will have no idea it's possible to do the conversion automatically.

screen shot 2017-07-11 at 13 45 27

Let's make sure that we preview changes to the docs in the future (by running the local server), and that new mentions of deprecations link to blog posts that link to codemods.

**Comment by gaearon** [comment]
(I merged a PR to fix this but let's keep in mind for future deprecations.)

**Pull request merged**